lib/amqp/client/connection.rb

```diff
@@ -400,6 +400,8 @@ def open_socket(host, port, tls, options)
           socket.post_connection_check(host) || raise(Error, "TLS certificate hostname doesn't match requested")
         end
         socket
+      rescue Errno::ECONNREFUSED => e
```

Comment:
Let's catch all systemcall errors (super class of `Errno::ECONNREFUSED` and others). Maybe `OpenSSL::Errors` too..  

```suggestion
      rescue SystemCallError => e
```